src/js/index.js

```diff
@@ -0,0 +1,114 @@
+import { ALERT_MESSAGE, OPERATION, DIGIT_NUMBER_MEX_LENGTH } from './constants.js';
+import { getSelector, addEvent } from './utils.js';
+import { getTotal, addDigitNumber, getTotalWithOperation, addOperation, getCalculateStep } from './domains.js';
+
+const initState = {
+  total: '0',
+  digits: [],
+  calculateStep: 0,
+  operators: [],
+};
+
+function App() {
+  // 상태값 정의
+  this.state = initState;
+
+  // 상태값 설정 함수
+  const setState = (newState) => {
+    this.state = { ...this.state, ...newState };
+    render();
+  };
+
+  // 변수
+  const totalEl = getSelector('#total');
+  const modifierEl = getSelector('.modifier');
+
+  // init 함수
+  this.init = () => {
+    initEventListeners();
+  };
+
+  // 렌더 함수
+  const render = () => {
+    totalEl.textContent = this.state.total;
+  };
+
+  // 상태값 초기화 함수
+  const resetState = () => {
+    this.state = initState;
+  };
```

Comment:
```suggestion
const initState = {
  total: '0',
  digits: [],
  calculateStep: 0,
  operators: [],
};

function App() {
  // 상태값 정의
  this.state = { ...initState };

  // 상태값 설정 함수
  const setState = (newState) => {
    this.state = { ...this.state, ...newState };
    render();
  };

  // 변수
  const totalEl = getSelector('#total');
  const modifierEl = getSelector('.modifier');

  // init 함수
  this.init = () => {
    initEventListeners();
  };

  // 렌더 함수
  const render = () => {
    totalEl.textContent = this.state.total;
  };

  // 상태값 초기화 함수
  const resetState = () => {
    this.state = { ...initState };
  };
```
일단 state에 initState를 바로 할당한 다음에 직접적으로 state를 수정할 경우 문제가 발생할 수도 있습니다.
![image](https://user-images.githubusercontent.com/18749057/158090374-abbd7b9a-cb73-4263-b69a-3cf5ade0a3dd.png)

코드 내에서 this.state에 직접 접근해서 수정하는 부분이 없어서 다행이긴 하지만.. 일단 잠재적인 위험이 있다는 점을 인지해주시면 좋을 것 같아요!